internal/oci/container.go

```diff
@@ -462,6 +462,9 @@ func (c *Container) pid() (int, error) {
 	if err := c.verifyPid(); err != nil {
 		return 0, err
 	}
+	if err := unix.Kill(c.state.InitPid, 0); errors.Is(err, unix.ESRCH) {
+		return 0, errors.Wrapf(err, "check whether %d is running", c.state.InitPid)
+	}
```

Comment:
Since `kill` is faster than `c.verifyPid()` I'd put it first.

OTOH if you want to check it as late as possible, leave it as is.

Reply:
waiting as long as possible allows us to *potentially* catch the case where the process has died but the proc entry hasn't been cleaned, which is most of the point of adding this kill. if we do it before, there's no way we'd catch it.

Reply:
I highly doubt that such situation (" the process has died but the proc entry hasn't been cleaned") is at all possible. Have you ever seen it?

Surely a process can disappear any time, say right after we read its /proc entry.

Adding yet another kill (with 0) to check if the process is still there before using this pid makes the race window a bit shorter but it does not close it. Instead we should either expect the pid to still be there (in most places), or disappear at any time (when stopping, like it is done in WaitContainerStop / Kill).

Anyway, syscalls are fast nowdays, I guess it won't hurt.

Reply:
I am not positive I've seen it, but before adding this kill, the window after running the kill  in the test needed to be higher, and adding the kill reduced the window. I believe it helped